### DISCUSSION TITLE
Best way to handle guarding routes based on user's permissions?

### DISCUSSION THREAD

**Discussion by peterchappy** [TanStack/router#226] [Q&A]
I'm curious with `react-location` what is the best way to handle dynamic routes based on the user's role? In `react-router` there is a few ways to peel that potato, but it's pretty straight forward given the nature of declaring routes as components. Considering that `react-location` constructs the router from an object I'm wonder what the community thinks the best way to handle this is as I don't see an example in the docs.

**Comment by Jared-Dahlke** [comment]
here's what i do, for what it's worth:
```
 {
                path: 'edit',
                children: [
                    {
                        path: ':scenarioId',
                        element: (
                            <ProtectedRoute
                                component={mycomponent}
                                permissionsNeeded={[perms.ADMIN_READ]}
                            />
                        )
                    }
                ]
            }

```
